parser/parser.go

```diff
@@ -345,3 +349,10 @@ func (p *Parser) wrapHTTP(directive *gonginx.Directive) *gonginx.HTTP {
 func (p *Parser) parseUpstreamServer(directive *gonginx.Directive) *gonginx.UpstreamServer {
 	return gonginx.NewUpstreamServer(directive)
 }
+
+func (p *Parser) Close() (err error) {
+	if p.file != nil {
+		err = p.file.Close()
+	}
+	return err
+}
```

Comment:
provide interface to Close parser without Parse file. (for case with 
`defer parser.Close()`)